src/User/UserRepository.php

```diff
@@ -26,14 +30,59 @@ public function getSpecialistId(int $userId): int
         return (int) $this->getUserProperty($userId, 'user_specialist_id');
     }
 
-    private function getUserProperty(int $userId, string $property): string
+    public function getNick(int $userId): ?string
+    {
+        return $this->getUserProperty($userId, 'user_nick');
+    }
+
+    private function getUserProperty(int $userId, string $property): ?string
     {
-        return $this->createQueryBuilder()
+        $data = $this->createQueryBuilder()
             ->select($property)
             ->from('users')
             ->where('user_id = :userId')
             ->setParameter('userId', $userId)
             ->execute()
             ->fetchOne();
+        return !is_bool($data) ? $data : null;
+    }
+
+    function count(): int
+    {
+        return (int) $this->createQueryBuilder()
+            ->select("COUNT(*)")
+            ->from('users')
+            ->execute()
+            ->fetchOne();
+    }
+
+    function countActiveSessions(int $timeout): int
+    {
+        return (int) $this->createQueryBuilder()
+            ->select('COUNT(*)')
+            ->from('user_sessions')
+            ->where('time_action > ?')
+            ->setParameter(0, time() - $timeout)
+            ->execute()
+            ->fetchOne();
+    }
+
+    public function cleanUpPoints(int $threshold = 0): int
+    {
+        // TODO
+        $cfg = Config::getInstance();
```

Comment:
Should try to avoid using code from the htdocs folder as those classes still use the non doctrine database connection

Reply:
Would you rewrite the method to provide the config value as function parameter, for the moment? (and update all relevant calls)?

Reply:
Would leave as is for now. Next step move Config and Log outside. Can for instance be in a service

Reply:
Yes, thought so as well.